app/controllers/api/services_controller.rb

```diff
@@ -55,28 +54,28 @@ def create
     creator = ServiceCreator.new(service: @service)
 
     if can_create? && creator.call(create_params)
-      flash[:notice] = t('flash.services.create.notice', resource_type: product_or_service_type)
+      flash[:notice] = t('flash.services.create.notice', resource_type: 'Product')
       redirect_to admin_service_path(@service)
     else
-      flash.now[:error] = @service.errors.full_messages.to_sentence.presence || I18n.t!('flash.services.create.errors.default', {resource_type: product_or_service_type})
+      flash.now[:error] = @service.errors.full_messages.to_sentence.presence || I18n.t!('flash.services.create.errors.default', {resource_type: 'Product'})
       activate_menu :dashboard
       render :new
     end
   end
 
   def update
     if integration_settings_updater_service.call(service_attributes: service_params, proxy_attributes: proxy_params)
-      flash[:notice] =  t('flash.services.update.notice', resource_type: product_or_service_type)
+      flash[:notice] =  t('flash.services.update.notice', resource_type: 'Product')
       redirect_back_or_to :action => :settings
     else
-      flash.now[:error] = t('flash.services.update.error', resource_type: product_or_service_type)
-      render action: params[:update_settings].present? ? settings_page : :edit # edit page is only page with free form fields. other forms are less probable to have errors
+      flash.now[:error] = t('flash.services.update.error', resource_type: 'Product')
```

Comment:
if it is always the same resource type, why we do not simply add the name directly in the string of the locale instead of passing the variable 'Product' from everywhere?

Reply:
Yeah, good catch 💪